sys/mips/include/sf_buf.h

```diff
@@ -39,7 +39,7 @@ sf_buf_kva(struct sf_buf *sf)
 	vm_page_t	m;
 
 	m = (vm_page_t)sf;
-	return (MIPS_PHYS_TO_DIRECT(VM_PAGE_TO_PHYS(m)));
+	return ((vm_ptr_t)MIPS_PHYS_TO_DIRECT(VM_PAGE_TO_PHYS(m)));
```

Comment:
This cast confuses me; is it not already a `uintptr_t`?

Reply:
I think it's a `caddr_t` since the global mips roots are `caddr_t` and not `uintcap_t`.  For RISC-V, I made `dmap_capability` a `void *`, but I cast it to `vm_pointer_t` when generating DMAP addresses so that the result of the DMAP macros are `vm_pointer_t` values.  I suspect MIPS might benefit from the same approach and require fewer diffs if `MIPS_PHYS_TO_DIRECT` returns `vm_pointer_t` and if it's inverse accepts a `vm_pointer_t`.